generator/src/main/java/com/scottlogic/deg/generator/generation/databags/DataBagValue.java

```diff
@@ -33,5 +36,17 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(value, format);
     }
+
+    public Object getValue() throws InvalidProfileException {
```

Comment:
So, formattedAs only applies to some output types, that feels wrong

Reply:
I disagree strongly. A field typed as a number shouldn't output as a string in a format that can represent numbers. `formattedAs` is only for formats that _have_ to output as string.

Reply:
(I do acknowledge there's a use case for "this field is a number represented as a string", though. In the short term, since this change alters the behaviour of the system in a fairly significant way, I'd say it shouldn't be part of a PR about cucumber testing)